src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/ConnectionPluginManager.java

```diff
@@ -47,18 +49,22 @@
   protected PropertySet propertySet = null;
   protected IConnectionPlugin headPlugin = null;
   ClusterAwareConnectionProxy proxy;
-  private final Thread shutdownHook = new Thread(this::releaseResources);
+  protected static final List<ConnectionPluginManager> instances = new CopyOnWriteArrayList<>();
```

Comment:
https://stackoverflow.com/questions/39573764/merging-list-concurrently-which-is-better-copyonwritearraylist-or-concurrentl
I suspect there will be way more writes than reads to this list so I propose we use ConcurrentLinkedQueue instead, what do you think?

Reply:
Sounds good. Will change this.